modules/handlers/follow_spot.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from telegram import Update
+from telegram.ext import CallbackContext
+from telegram.error import Unauthorized
+from modules.utils import EventInfo
+from modules.data import Config
+from modules.data.db_manager import DbManager
+
+
+logger = logging.getLogger(__name__)
+
+def follow_spot_callback(update: Update, context: CallbackContext) -> int:
+    """Handles the follow callback.
+
+    Args:
+        update: update event
+        context: context passed by the handler
+
+    Returns:
+        next state of the conversation
+    """
+    info = EventInfo.from_callback(update, context)
+    message_id = info.message.reply_to_message.message_id if Config.meme_get('comments') else info.message_id
+
+    # Check if user is already following this spot
+    is_following: bool = len(DbManager.select_from(
+        table_name = "user_follow",
+        select = "*",
+        where = "user_id = %s and message_id = %s",
+        where_args = (info.user_id, message_id)
+    )) > 0
+
+    if is_following:
+        answer_text = "Non stai più seguendo questo spot."
+        # Forget the stored data
+        DbManager.delete_from(
+            table_name="user_follow",
+            where="user_id = %s and message_id = %s",
+            where_args=(info.user_id, message_id)
+        )
+        
+        info.bot.send_message(
+            chat_id = info.user_id,
+            text = "Non stai più seguendo questo spot.",
+            reply_to_message_id = message_id,
+            disable_notification = True
+        )
+    else:
+        answer_text = "Stai seguendo questo spot."
```

Comment:
you could add something like 
```suggestion
        post_url = f"https://t.me/c/{str(info.chat_id).replace('-100', '')}/{info.message_id}"
        answer_text = f"Stai seguendo questo spot. {post_url}"
```

You could also replace that text URL into an icon ➡️  with the href="post_url"...
or put a telegram button with the arrow

Reply:
p.s. you could take the info.chat_id variable from settings.yaml `channel_group_id`, but it's the same